### ISSUE TITLE
Lokala notifikationer

### ISSUE THREAD

**Issue by viktorlarsson** [kolplattformen/skolplattformen#187]
Eftersom vi inte har tillgång till någon form av backend så finns det ett sätt att planera upp notifikationer lokalt med hjälp av `react-native-push-notification`

Jag har implementerat detta i 2 olika appar i produktion, så har rätt bra kunskap samt en del kod att implementera och tar gärna an mig uppgiften men vill lägga det som diskussion, vet att ni redan har att man sparar till kalendern.

Lite gotchas:
- Går inte att fånga om något i kalendern har blivit inställd förens man faktiskt öppnar appen

**Comment by ErikHellman** [comment]
Intressant idé! Idag så sparas inget lokalt i appen, så det blir svårt att få till detta på ett bra sätt. Vi vill undvika att göra något sådant idag med anledningen att det kan missuppfattas som att vi skulle spara personuppgifter. 

Vill jättegärna ha en sådan funktion, men just nu får vi nog avvakta.

**Comment by viktorlarsson** [comment]
Bra fångat! Det har du alldeles rätt i.

Jag har iaf påbörjat lägga in på en fork och en branch ifall det skulle bli aktuellt

https://github.com/viktorlarsson/skolplattformen/tree/feature/local-notifications/packages/app/services

**Comment by viktorlarsson** [comment]
On another note, kan man joina Discorden ifall man ska hugga något annat @ErikHellman ?

**Comment by stefannilsson** [comment]
Schemalagda API-polls (beroende på hur mycket de nu lastar Skolplatformens backend samt batteri) skulle kanske kunna kombineras med `react-native-push-notification`?

_State_ man då skulle behöva hålla i appen blir då "nuläge" (IDn/GUID'ar/Statusar == Inga personuppgifter) för att kunna jämföra om något förändrats och då notifiera?

**Comment by viktorlarsson** [comment]
Background tasks i iOS är nyckfulla och man vet inte när de körs. De körs dessutom med lägre tidsintervall desto mindre appen används om det var det du tänkte?

I ett av projekten har vi löst det med att aldrig specifiera vad push notisen handlar om, vilket man skulle kunna göra här. Sämre användarupplevelse men 

”Du har en uppkommande kalenderhändelse i appen, logga in och kolla”

**Comment by stefannilsson** [comment]
Ja, det blir väl tyvärr trade-off'en om man inte får ha något backend som håller koll.
Frågan är om man hellre är helt utan notiser som användare om man inte kan förlita sig på att man faktiskt blir notifierad (varje gång, och med någon rimlig delay)? Eller om man heller får en notis ibland över förändringar som är värda att notifiera.

Man behöver väl ändå lagra något i app'en för att kunna inse att "något allmänt" har hänt.
